### PR TITLE
Add initial implementation of <vaadin-month-calendar>

### PR DESCRIPTION
**Pull request by tehapo**
Alternative version to https://github.com/vaadin/vaadin-date-picker/pull/7 implemented with `dom-repeat` templates.

Connects to #4  

[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-date-picker/9)

### CODE REVIEW

**Comment by tomivirkki** [comment]
Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-month-calendar.html, line 36 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XOYd02jc6o1jnOBwi:-K9XOYd02jc6o1jnOBwj:-513192568) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L36)):*
This looks nasty.
You can get around the issue with inline-block elements by using
...div {
float: left;
height: XXpx;
...
}

---

*[vaadin-month-calendar.html, line 80 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XNU26A4ntx_Rq8qgY:-K9XNU26A4ntx_Rq8qgZ:-747134556) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L80)):*
Why not inline this like _getWeekDayNames? The property isn't needed

---

*[vaadin-month-calendar.html, line 90 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XN5CGOtjHdnSRa6MQ:-K9XN5CGOtjHdnSRa6MR:801544732) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L90)):*
Date is never falsey so you could shorten this to
if (date1 && date2) 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9)*

**Comment by tehapo** [comment]
Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-month-calendar.html, line 36 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XOYd02jc6o1jnOBwi:-K9XRleU53EnIVtdfuo9:-1207060872) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L36)):*
Trying to keep away from floating events whenever possible. I'll look into this tomorrow.

---

*[vaadin-month-calendar.html, line 80 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XNU26A4ntx_Rq8qgY:-K9XRh0LiCuPONQzaLVI:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L80)):*
Done.

---

*[vaadin-month-calendar.html, line 90 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XN5CGOtjHdnSRa6MQ:-K9XSJiySTg7yaV9iejl:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L90)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9)*

**Comment by tehapo** [comment]
Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-month-calendar.html, line 36 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XOYd02jc6o1jnOBwi:-K9XSPAIRuzZymH7d8mN:-290674602) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L36)):*
Floating *elements* :smile: 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9)*

**Comment by tomivirkki** [comment]
*[vaadin-month-calendar.html, line 36 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XOYd02jc6o1jnOBwi:-K9XWTv90k5oAOn3y7y6:1261274058) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L36)):*
Are there any benefits in not using float in this case? Especially if the drawback is that you need to one-line your source html.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9)*

**Comment by jounik** [comment]
Why not use flexbox instead of inline-block?

**Comment by tomivirkki** [comment]
Or display: table; https://benfrain.com/css-performance-test-flexbox-v-css-table-fight/

**Comment by tehapo** [comment]
Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-month-calendar.html, line 36 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9#-K9XOYd02jc6o1jnOBwi:-K9XbNOntaP1GtFJ3JYY:1231986550) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/bc0cc8e42f5c7a34b578d8dcd90de002d712eed0/vaadin-month-calendar.html#L36)):*
Floating always leads to the need to clear the floats at some point to make the containing element stretch with the content. Anyways, you're right that the template was ugly as hell without whitespace and it's now fixed with using flexbox.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/9)*
